libs/generic-view/store/src/lib/imports/contacts-mappers/vcard/map-vcard.ts

```diff
@@ -0,0 +1,152 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import vCard from "vcf"
+import { UnifiedContact } from "device/models"
+import { getDisplayName } from "../helpers"
+import { first, isArray, isEmpty, last } from "lodash"
+
+export type jCard = ReturnType<vCard["toJSON"]>
+
+export const mapVcard = (data: jCard[]): UnifiedContact[] => {
+  return data
+    .map((jCard, index) => {
+      const id = getId(jCard) || `${index}`
+      const nameFields = getN(jCard)
+      const note = getNote(jCard)
+      const nickName = getNickname(jCard)
+      return {
+        id,
+        ...nameFields,
+        ...(nickName && { nickname: nickName }),
+        displayName: getDisplayName(nameFields),
+        phoneNumbers: getTel(jCard).filter(({ value }) => value),
+        emailAddresses: getEmail(jCard).filter(({ value }) => value),
+        addresses: getAdr(jCard).filter(({ type, ...item }) => !isEmpty(item)),
+        organizations: getOrg(jCard).filter((item) => !isEmpty(item)),
+        urls: getUrl(jCard).filter(({ value }) => value),
+        ...(note && { note }),
+      }
+    })
+    .filter(
+      ({ id, displayName, ...item }) => !Object.values(item).every(isEmpty)
+    )
+}
+
+const getFields = (item: jCard, key: string) => {
+  const fields = item[1].filter((property) => property[0] === key)
+  return fields
+    .map((field) => {
+      const type = isArray(field[1].type) ? last(field[1].type) : field[1].type
+      const value = field[3]
+      return { type, value }
+    })
+    .filter(({ value }) => value)
+}
+
+const getId = (item: jCard) => {
+  const field = first(getFields(item, "uid"))
+  return field?.value as string | undefined
+}
+
+const getN = (
```

Comment:
I think we need a more explicit name for this function

Reply:
Changed